File: src/main/java/net/lomeli/trophyslots/core/command/CommandRemoveSlots.java
package net.lomeli.trophyslots.core.command;

import net.minecraft.command.CommandBase;
import net.minecraft.command.ICommandSender;
import net.minecraft.entity.player.EntityPlayerMP;
import net.minecraft.util.ChatComponentText;
import net.minecraft.util.ChatComponentTranslation;
import net.minecraft.util.StatCollector;

import net.lomeli.trophyslots.TrophySlots;
import net.lomeli.trophyslots.core.SlotUtil;
import net.lomeli.trophyslots.core.network.MessageSlotsClient;

public class CommandRemoveSlots extends CommandBase {
    @Override
    public String getCommandName() {
        return "remove-slots";
    }

    @Override
    public int getRequiredPermissionLevel() {
        return 2;
    }

    @Override
    public String getCommandUsage(ICommandSender sender) {
        return "command.trophyslots.remove-slots.usage";
    }

    @Override
    public void processCommand(ICommandSender sender, String[] args) {
        if (args.length == 2 || args.length == 3) {
            EntityPlayerMP player = args.length == 2 ? getCommandSenderAsPlayer(sender) : getPlayer(sender, args[1]);
            int rmSlots = args.length == 2 ? parseString(args[1]) : parseString(args[2]);
            if (player != null && (rmSlots > 0 && rmSlots <= SlotUtil.getMaxSlots())) {
                int slots = SlotUtil.getSlotsUnlocked(player);
                if (slots <= 0)
                    sender.addChatMessage(new ChatComponentText(String.format(StatCollector.translateToLocal("command.trophyslots.remove-slots.error.none"), player.getDisplayName())));
                else {
                    slots -= rmSlots;
                    if (slots < 0)
                        slots = 0;
                    SlotUtil.setSlotsUnlocked(player, slots);
                    TrophySlots.packetHandler.sendTo(new MessageSlotsClient(slots), player);
                    player.addChatMessage(new ChatComponentText(String.format(StatCollector.translateToLocal("msg.trophyslots." + (slots == 0 ? "lostAll" : "lostSlot")), rmSlots)));
                    sender.addChatMessage(new ChatComponentText(String.format(StatCollector.translateToLocal("command.trophyslots.remove-slots.success"), rmSlots, player.getDisplayName())));
                }
            } else
                sender.addChatMessage(new ChatComponentText(String.format(StatCollector.translateToLocal("command.trophyslots.remove-slots.error"), SlotUtil.getMaxSlots())));
        } else
            sender.addChatMessage(new ChatComponentTranslation(getCommandUsage(sender)));
    }

    private int parseString(String st) {
        try {
            return Integer.parseInt(st);
        } catch (Exception e) {
            return -1;
        }
    }
}

File: src/main/kotlin/net/lomeli/trophyslots/core/command/CommandRemoveSlots.kt
package net.lomeli.trophyslots.core.command

import net.lomeli.trophyslots.TrophySlots
import net.lomeli.trophyslots.capabilities.slots.SlotManager
import net.lomeli.trophyslots.core.network.MessageSlotsClient
import net.minecraft.command.CommandBase
import net.minecraft.command.ICommandSender
import net.minecraft.entity.player.EntityPlayerMP
import net.minecraft.server.MinecraftServer
import net.minecraft.util.text.TextComponentString
import net.minecraft.util.text.TextComponentTranslation
import net.minecraft.util.text.translation.I18n

class CommandRemoveSlots : CommandBase() {
    override fun execute(server: MinecraftServer?, sender: ICommandSender?, args: Array<out String>?) {
        if (sender != null) {
            if (args != null && (args.size == 2 || args.size == 3)) {
                val player: EntityPlayerMP
                val rmSlots: Int
                if (args.size == 3) {
                    rmSlots = parseInt(args[2])
                    player = CommandBase.getPlayer(server, sender, args[1])
                } else {
                    rmSlots = parseInt(args[1])
                    player = CommandBase.getCommandSenderAsPlayer(sender)
                }
                val slotInfo = SlotManager.getPlayerSlotInfo(player)!!
                if (player != null && (rmSlots > 0 && rmSlots <= slotInfo.getMaxSlots())) {
                    var slots = slotInfo.getSlotsUnlocked()
                    if (slots <= 0)
                        sender.sendMessage(TextComponentString(I18n.translateToLocal("command.trophyslots.remove_slots.error.none").format(player.displayName.unformattedText)))
                    else {
                        slots -= rmSlots
                        if (slots < 0)
                            slots = 0
                        slotInfo.setSlots(slots)
                        TrophySlots.packetHandler?.sendTo(MessageSlotsClient(slots), player)
                        if (slots == 0)
                            player.sendMessage(TextComponentTranslation("msg.trophyslots.lost_all"))
                        else
                            player.sendMessage(TextComponentString(I18n.translateToLocal("msg.trophyslots.lost_slot").format(rmSlots)))
                        sender.sendMessage(TextComponentString(I18n.translateToLocal("command.trophyslots.remove_slots.success").format(rmSlots, player.displayName.unformattedText)))
                    }
                } else
                    sender.sendMessage(TextComponentString(I18n.translateToLocal("command.trophyslots.remove_slots.error").format(slotInfo.getMaxSlots())))
            } else
                sender.sendMessage(TextComponentTranslation(getUsage(sender)))
        }
    }

    override fun getName(): String? = "remove-slots"

    override fun getUsage(sender: ICommandSender?): String? = "command.trophyslots.remove_slots.usage"

    override fun getRequiredPermissionLevel(): Int = 2
}